GLMakie/src/screen.jl

```diff
@@ -663,10 +667,45 @@ function fps_renderloop(screen::Screen)
     end
 end
 
+
+function requires_update(screen::Screen)
+    for (_, _, robj) in screen.renderlist
+        visible = Bool(to_value(get(robj.uniforms, :visible, true)))
```

Comment:
I think we may want to use this PR to change the visible field into a well typed, mandatory field for RenderObject.

Reply:
Sounds good.

Reply:
Not sure if I did what you had in mind but I moved it to a field.

I also had the logic for re-rendering with visible wrong. I tried to set it up in a way that invisible renderobjects can't trigger updates, but the way I did it caused `p.visible[] = false` to not trigger a re-render. I fixed that by moving the check to the renderobject updater functions.